e2e/main_test.go

```diff
@@ -17,47 +17,94 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// Integration tests are run against memstore whereas
-// testnet integration tests are run against eigenda backend talking to testnet disperser.
-// Some of the assertions in the tests are different based on the backend as well.
-// e.g, in TestProxyServerCaching we only assert to read metrics with EigenDA
-// when referencing memstore since we don't profile the eigenDAClient interactions
+// TestType is an enum describing various types of tests. This TestType is used to determine whether
+// a given test should run in the configured test environment.
+type TestType int
+
+const (
+	// StandardIntegration is a test that should run in any integration environment (local, or testnet)
+	StandardIntegration TestType = iota
+	// LocalOnlyIntegration is a test that should run only in a local integration environment, NOT on testnet
+	LocalOnlyIntegration
+	// Fuzz is a fuzz test
+	Fuzz
+)
+
 var (
+	config testFlagConfig
+)
+
+// testFlagConfig contains the boolean flags used to configure test execution
+type testFlagConfig struct {
 	runTestnetIntegrationTests bool // holesky tests
-	runIntegrationTests        bool // memstore V1 tests
-	runIntegrationTestsV2      bool // memstore V2 tests
+	runLocalIntegrationTests   bool // memstore tests
 	runFuzzTests               bool // fuzz tests
-)
+	enableV2                   bool
+}
 
+// validate checks that the values in testFlagConfig are sensical, and prints the configured values for convenience
+func (tfc *testFlagConfig) validate() {
+	if tfc.runLocalIntegrationTests && tfc.runTestnetIntegrationTests {
+		panic("only one of LOCAL=true or TESTNET=true env var can be set")
+	}
+
+	fmt.Printf(
+		"runFuzzTests: %v, runLocalIntegrationTests: %v, runTestnetIntegrationTests: %v, enableV2: %v\n",
+		tfc.runFuzzTests, tfc.runLocalIntegrationTests, tfc.runTestnetIntegrationTests, tfc.enableV2,
+	)
+}
+
+// flagActivated returns true if a given environment variable is active, otherwise false
 func flagActivated(envVar string) bool {
 	return os.Getenv(envVar) == "true" || os.Getenv(envVar) == "1"
 }
 
+// shouldRunTest accepts a parameter testType, which represents the type of test that is calling shouldRunTest. Based
+// on the testType, this method returns whether the test in question should run based on the configured values.
+func shouldRunTest(testType TestType) bool {
+	switch testType {
+	case StandardIntegration:
+		return config.runLocalIntegrationTests || config.runTestnetIntegrationTests
```

Comment:
feels to me like this is basically saying "skip this test when running fuzz tests"?

Can we instead just move fuzz tests to their own standalone package. I think (?) all the fuzz tests are just meant to be independent anyways? Would reduce the complexity of understanding when a test is skipped and why.

Reply:
Fuzz tests moved to a different package